components/automate-cli/pkg/verifyserver/services/externalpostgresqlservice/externalpostgresqlservice.go

```diff
@@ -0,0 +1,121 @@
+package externalpostgresqlservice
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/db"
+	"github.com/chef/automate/lib/logger"
+	_ "github.com/lib/pq"
+)
+
+type ISExternalPostgresqlService interface {
+	GetPgConnection(*models.ExternalPgRequest) (*models.ExternalPgResponse, error)
+}
+
+type ExternalPostgresqlService struct {
+	logger  logger.Logger
+	DBUtils db.DB
+	Req     *models.ExternalPgRequest
+}
+
+func NewExternalPostgresqlService(db db.DB, logger logger.Logger) ISExternalPostgresqlService {
+	return &ExternalPostgresqlService{
+		logger:  logger,
+		DBUtils: db,
+	}
+}
+
+func (pg *ExternalPostgresqlService) GetPgConnection(req *models.ExternalPgRequest) (*models.ExternalPgResponse, error) {
+	pg.Req = req
+
+	//creating a temp file to the copy content of a rootcert into a file
+	rootcert, err := createTempFile(req.PostgresqlRootCert)
+	if err != nil {
+		return nil, err
+	}
+
+	err = pg.CheckExternalPgConnection(rootcert)
+	var resp *models.ExternalPgResponse
+
+	if err != nil {
+		pg.logger.Error(err)
+		resp = &models.ExternalPgResponse{
+			Passed: false,
+			Checks: []models.ExternalPgConnectionDetails{failResponse(constants.EXTERNALPGFAILCONNECTIONTITLE, constants.EXTERNALPGCONNECTIONERRORMSG, constants.EXTERNALPGCONNECTIONRESOLUTIONMSG)},
+		}
+	}
+
+	resp = &models.ExternalPgResponse{
+		Passed: true,
+		Checks: []models.ExternalPgConnectionDetails{successResponse(constants.EXTERNALPGSUCCESSCONNECTIONTITLE, constants.EXTERNALPGCONNECTIONSUCCESSMSG)},
+	}
+
+	// delete the file when done
+	deleteTempFile(rootcert)
+
+	return resp, nil
+}
+func (p *ExternalPostgresqlService) CheckExternalPgConnection(rootcert string) error {
+
+	connStr := fmt.Sprintf("host=%s port=%s user=%s "+
+		"password=%s dbname=postgres sslmode=verify-ca sslrootcert=%s",
+		p.Req.PostgresqlInstanceUrl, p.Req.PostgresqlInstancePort, "%s", "%s", rootcert)
+
+	superUserConnStr := fmt.Sprintf(connStr, p.Req.PostgresqlSuperUserUserName, p.Req.PostgresqlSuperUserPassword)
+	DbUserConnStr := fmt.Sprintf(connStr, p.Req.PostgresqlDbUserUserName, p.Req.PostgresqlDbUserPassword)
+
+	QueryStr := []string{superUserConnStr, DbUserConnStr}
+
+	for i := 0; i < len(QueryStr); i++ {
+		err := p.DBUtils.InitPostgresDB(QueryStr[i])
+		if err != nil {
+			p.logger.Error("External Postgresql Connection failed: ", err.Error())
+			return err
+		}
+		p.logger.Info("External Postgresql aws connection success")
+	}
+	return nil
+}
+
+func createTempFile(content string) (string, error) {
+
+	tempFile, err := ioutil.TempFile("", "root-cert") // nosemgrep
```

Comment:
we can pass err from line 87 to the return statement at line 89

Reply:
done